### PR TITLE
[gbc] Use latest stack

### PR DESCRIPTION
**Pull request by ara-ayvazyan**

### CODE REVIEW

**Comment by ara-ayvazyan** [comment]
I've canceled the CI builds intentionally since the image is not updated.

**Comment by ara-ayvazyan** [comment]
Looks like this did not actually fix the issue 😞

File: compiler/stack_build.cmake [hunk 1]
```diff
@@ -11,7 +11,7 @@
 cmake_policy (SET CMP0012 NEW)
 
 execute_process (
-    COMMAND ${STACK_EXECUTABLE} setup
+    COMMAND ${STACK_EXECUTABLE} setup --allow-different-user
```
Comment (chwarr) [inline compiler/stack_build.cmake]: It looks to me that this is solving a problem specific to our use of Travis CI + a Docket container. It doesn't seem appropriate to pass `--allow-different-user` unconditionally here: for developer/customer machines that isn't the right default.

Is there a way we can fix this just from the Travis build script?

Comment (ara-ayvazyan) [reply compiler/stack_build.cmake]: That flag is no more needed.

Comment (ara-ayvazyan) [reply compiler/stack_build.cmake]: I've sent separate PRs #681 and #682.

**Pull request merged**